src/components/landingPage/upload.tsx

```diff
@@ -1,218 +1,123 @@
-          style={{ width: 200 }}
-          size="small"
-          disabled={!locationOptions}
-        >
-          {locationOptions &&
-            locationOptions.map((loc: string) => (
-              <MenuItem value={loc}>{loc}</MenuItem>
-            ))}
-        </Select>
-      </FormControl>
+          Select tree JSON <input hidden type="file" />
+        </Button>
+        <FormHelperText>
+          This must be in Nextstrain's JSON format{" "}
+          <Tooltip
+            componentsProps={tooltipProps}
+            title={
+              <>
+                If you need help generating a phylogenetic tree, head over to{" "}
+                <a href="https://czgenepi.org" style={{ color: "white" }}>
+                  CZ GEN EPI
+                </a>
+              </>
+            }
+          >
+            <InfoOutlined
+              color="primary"
+              style={{ fontSize: 16, position: "relative", top: 4 }}
+            />
+          </Tooltip>
+        </FormHelperText>
+      </p>
       <p>
-        <b> 3. Optionally, upload sample metadata (e.g., a line list)</b>
-        <br />
-        <i>
-          Helps you identify meaningful clusters and integrate genomic and
-          epidemiological insights.
-        </i>
-        <br />
-        <input
-          type="file"
-          name="file"
-          onChange={(event: any) => {
-            handleMetadataUpload(event.target.files[0]);
+        <Button
+          variant="contained"
+          disabled={!state.tree}
+          disableElevation
+          disableRipple
+          onClick={() => {
+            dispatch({ type: "upload modal toggled" });
           }}
-        />
-      </p>
-
-      <FormControl>
-        <FormLabel style={{ width: "30em" }}>
-          Select field/column containing names to match metadata rows to samples
-          in the tree.
-        </FormLabel>
-        <Select
-          id="metadataKeySelect"
-          //@ts-ignore
-          onChange={(e) =>
-            dispatch({
-              type: "metadata match field selected",
-              data: e.target.value,
-            })
-          }
-          style={{ width: 200 }}
-          size="small"
-          disabled={!metadataKeyOptions}
+          size="large"
         >
-          {metadataKeyOptions &&
-            metadataKeyOptions.map((key: string) => (
-              <MenuItem value={key}>{key}</MenuItem>
-            ))}
-        </Select>
-      </FormControl>
-      <br />
-      <br />
-      <Button
-        variant="contained"
-        disableElevation
-        disableRipple
-        onClick={(e) => {
-          dispatch({ type: "upload submit button clicked" });
-          navigate("/galago/app");
-        }}
-        disabled={!state.division || !state.location || !state.tree}
+          Get started
+        </Button>
+      </p>
+      <Dialog
+        open={state.uploadModalOpen}
+        onClose={() => dispatch({ type: "upload modal toggled" })}

```

Comment:
Nit: I'm torn about this one. If it's explicitly an `onClose` event, it feels like "toggle" is not really the right word. It's more of an `upload modal closed`. The upside of doing it with a toggle is you can save on reducer funcs, you just need one that flip-flops. But if you ever hit a case where multiple pieces of logic attempt to simultaneously close something (or the user can double-click fast), you can get weird behavior where even/odd starts to matter although it shouldn't. I generally prefer an explicit `open` and `close` event. Although if you also have a button that does genuinely toggle some other display, I stop having even mild opinions. If your other `Button` above is really a toggle, then I'd say leave it as is, but if the `Button` is really the "open" button, I think having two differing events is the slightly better -- although not that important -- way to do things.

Reply:
Good style tip, thanks!